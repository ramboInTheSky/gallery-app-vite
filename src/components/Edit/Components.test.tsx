/** @jsxImportSource @emotion/react */
import { render } from "@testing-library/react";
import {
  Container,
  Form,
  Label,
  Input,
  RangeInput,
  PreviewContainer,
  Button,
} from "./Components";

describe("Styled Components Snapshots", () => {
  it("renders Container", () => {
    const { asFragment } = render(<Container>Content</Container>);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders Form", () => {
    const { asFragment } = render(<Form>Form Content</Form>);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders Label", () => {
    const { asFragment } = render(
      <Label>
        Label Text <Input />
      </Label>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders Input", () => {
    const { asFragment } = render(<Input />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders RangeInput", () => {
    const { asFragment } = render(<RangeInput type="range" />);
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders PreviewContainer", () => {
    const { asFragment } = render(
      <PreviewContainer>
        <img src="example.jpg" alt="Preview" />
      </PreviewContainer>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it("renders Button", () => {
    const { asFragment } = render(<Button>Click Me</Button>);
    expect(asFragment()).toMatchSnapshot();
  });
});
