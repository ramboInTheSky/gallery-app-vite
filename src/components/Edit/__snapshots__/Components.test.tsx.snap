// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Styled Components Snapshots renders Button 1`] = `
<DocumentFragment>
  .emotion-0 {
  padding: 0.75rem 1rem;
  border: none;
  background: var(--orange);
  color: white;
  border-radius: 4px;
  cursor: pointer;
  margin-left: 1rem;
  font-weight: 700;
}

.emotion-0:hover {
  opacity: 0.8;
}

<button
    class="emotion-0"
  >
    Click Me
  </button>
</DocumentFragment>
`;

exports[`Styled Components Snapshots renders Container 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin: 0 auto;
  max-width: 600px;
  min-height: 630px;
  color: var(--blue);
  border: 1px solid var(--purple);
  padding: 1.5rem 2rem 1rem 2rem;
  border-radius: 25px;
}

.emotion-0::before {
  content: "";
  position: absolute;
  top: 0;
  left: 0;
  width: 100%;
  height: 100%;
  opacity: 0.1;
  background: var(--light-purple);
  z-index: -1;
}

@media (max-width: 900px) {
  .emotion-0 {
    border: none;
  }
}

<div
    class="emotion-0"
  >
    Content
  </div>
</DocumentFragment>
`;

exports[`Styled Components Snapshots renders Form 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  gap: 1rem;
  color: var(--light-purple);
  margin-bottom: 2rem;
}

<div
    class="emotion-0"
  >
    Form Content
  </div>
</DocumentFragment>
`;

exports[`Styled Components Snapshots renders Input 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin-left: 1rem;
  padding: 0.5rem;
  border: 1px solid #ccc;
  border-radius: 4px;
  color: var(--purple);
}

<input
    class="emotion-0"
  />
</DocumentFragment>
`;

exports[`Styled Components Snapshots renders Label 1`] = `
<DocumentFragment>
  .emotion-0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  justify-content: space-between;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-1 {
  margin-left: 1rem;
  padding: 0.5rem;
  border: 1px solid #ccc;
  border-radius: 4px;
  color: var(--purple);
}

<label
    class="emotion-0"
  >
    Label Text 
    <input
      class="emotion-1"
    />
  </label>
</DocumentFragment>
`;

exports[`Styled Components Snapshots renders PreviewContainer 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin-bottom: 1.5rem;
  text-align: center;
}

.emotion-0 img {
  max-width: 100%;
  max-height: 50vh;
  min-height: 400px;
  border: 1px solid #ccc;
  border-radius: 8px;
}

<div
    class="emotion-0"
  >
    <img
      alt="Preview"
      src="example.jpg"
    />
  </div>
</DocumentFragment>
`;

exports[`Styled Components Snapshots renders RangeInput 1`] = `
<DocumentFragment>
  .emotion-0 {
  width: 85%;
  color: var(--orange);
}

<input
    class="emotion-0"
    type="range"
  />
</DocumentFragment>
`;
